request.go

```diff
@@ -508,3 +520,212 @@ func encodeSQLBind(enc *msgpack.Encoder, from interface{}) error {
 	}
 	return nil
 }
+
+// Request is an interface that provides the necessary data to create a request
+// that will be sent to a tarantool instance.
+type Request interface {
+	// Code returns a IPROTO code for the request.
+	Code() int32
+	// BodyFunc returns a functions that can fill an encoder with
+	// the request body or it returns an error if unable to create the function.
+	BodyFunc(resolver SchemaResolver) (func(enc *msgpack.Encoder) error, error)
+}
+
+type baseRequest struct {
+	requestCode int32
+}
+
+// Code returns a IPROTO code for the request.
+func (req *baseRequest) Code() int32 {
+	return req.requestCode
+}
+
+type spaceRequest struct {
+	baseRequest
+	space interface{}
+}
+
+func (req *spaceRequest) setSpace(space interface{}) {
+	req.space = space
+}
+
+type spaceIndexRequest struct {
+	spaceRequest
+	index interface{}
+}
+
+func (req *spaceIndexRequest) setIndex(index interface{}) {
+	req.index = index
+}
+
+// SelectRequest allows you to create a select request object for execution
+// by a Connection.
+type SelectRequest struct {
+	spaceIndexRequest
+	offset, limit, iterator uint32
+	key                     interface{}
+}
+
+// NewSelectRequest returns a new empty SelectRequest.
+func NewSelectRequest(space interface{}) *SelectRequest {
+	req := new(SelectRequest)
+	req.requestCode = SelectRequestCode
+	req.setSpace(space)
+	req.key = []interface{}{}
+	return req
+}
+
+// Index sets the index for the select request.
+// Note: default value is 0.
+func (req *SelectRequest) Index(index interface{}) *SelectRequest {
+	req.setIndex(index)
+	return req
+}
+
+// Offset sets the offset for the select request.
+// Note: default value is 0.
+func (req *SelectRequest) Offset(offset uint32) *SelectRequest {
+	req.offset = offset
+	return req
+}
+
+// Limit sets the limit for the select request.
+// Note: default value is 0.
```

Comment:
AFAIU we send 2^32-1 as the limit to the protocol if the limit is not sent on the client side. How it works here? Zero is interpreted as the alias for 2^32-1? A user can't pass zero to verify that the space is exists and accessible?

Reply:
Thank you, replaced to 0xFFFFFFFF by default.